src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -222,6 +224,34 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <!-- To achieve incremental build with property change. When any property changes, WriteOnlyWhenDifferent will be triggered to write cache file.
+  And the cache file's timestamp will be later, and it then triggers the incremental build.-->
+  <Target Name="_GenerateRuntimeConfigurationFilesInputCache" DependsOnTargets="ResolveAssemblyReferences">
```

Comment:
need discussion. maybe over kill, but that should be the "right thing"

Reply:
Seems OK. 

But why does this depend on ResolveAssemblyReferences? I don't see anything from what it does being used here.

runtimeconfig is so small, that I sort of wonder if it wound't be simpler, easier to maintain, or more efficient to just generate it and write the whole file only if different rather than any hashing that might need to be kept up to date with new properties.

Reply:
Good idea. The hash is probably bigger than the file. I reverted that commit and added https://github.com/dotnet/sdk/pull/4063/commits/31cec9dcea3ada9732bda2e46f64f6e997d17c90 instead.